src/fmu/dataio/_products/inplace_volumes.py

```diff
@@ -51,23 +48,48 @@ class InplaceVolumesResult(RootModel):
     root: List[InplaceVolumesResultRow]
 
 
-class InplaceVolumesJsonSchema(GenerateJsonSchema):
-    """Implements a schema generator so that some additional fields may be added."""
-
-    def generate(
-        self,
-        schema: Mapping[str, Any],
-        mode: Literal["validation", "serialization"] = "validation",
-    ) -> dict[str, Any]:
-        json_schema = super().generate(schema, mode=mode)
-        json_schema["$schema"] = self.schema_dialect
-        json_schema["$id"] = SCHEMA
-        json_schema["version"] = VERSION
-
-        return json_schema
-
-
-def dump() -> dict[str, Any]:
-    return InplaceVolumesResult.model_json_schema(
-        schema_generator=InplaceVolumesJsonSchema
-    )
+class InplaceVolumesSchema(SchemaBase):
+    VERSION: str = "0.1.0"
+    FILENAME: str = "inplace_volumes.json"
+    PATH: Path = FmuSchemas.PATH / "file_formats" / VERSION / FILENAME
+
+    class InplaceVolumesGenerateJsonSchema(GenerateJsonSchema):
+        """Implements a schema generator so that some additional fields may be added."""
```

Comment:
In a follow-up PR I think I want to see if I can apply a generator schema generator from `SchemaBase` to use as a default